### ISSUE TITLE
Stack overflow when migrating from 1.6.4 to 1.6.5

### ISSUE THREAD

**Issue by mihxil** [jlink/jqwik#327]
## Problem

I tried to up the jqwik version from 1.6.4 to 1.6.5. Which results in stack overflows of this kind:
```
[INFO] Running org.meeuw.physics.UnitsTest
Error:  Tests run: 39, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.723 s <<< FAILURE! - in org.meeuw.physics.UnitsTest
Error:  org.meeuw.physics.UnitsTest.multiplicativeGroupOperators(MultiplicativeGroup)  Time elapsed: 0.035 s  <<< ERROR!
java.lang.StackOverflowError
	at net.jqwik.engine.support.types.TypeUsageImpl.canBeAssignedTo(TypeUsageImpl.java)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:637)
	at net.jqwik.engine.support.types.TypeUsageImpl.canBeAssignedToUpperBounds(TypeUsageImpl.java:429)
	at net.jqwik.engine.support.types.TypeUsageImpl.lambda$canBeAssignedToUpperBounds$3(TypeUsageImpl.java:427)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:637)
	at net.jqwik.engine.support.types.TypeUsageImpl.canBeAssignedToUpperBounds(TypeUsageImpl.java:427)
	at net.jqwik.engine.support.types.TypeUsageImpl.canBeAssignedTo(TypeUsageImpl.java:401)
	at net.jqwik.engine.support.types.TypeUsageImpl.allTypeArgumentsCanBeAssigned(TypeUsageImpl.java:453)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:5[84](https://github.com/mihxil/math/runs/6240335463?check_suite_focus=true#step:5:84))
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:5[90](https://github.com/mihxil/math/runs/6240335463?check_suite_focus=true#step:5:90))
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
	at net.jqwik.engine.support.types.TypeUsageImpl.matchesWithTypeArguments(TypeUsageImpl.java:590)
```

See: https://github.com/mihxil/math/runs/6240335463?check_suite_focus=true in which it is contained.

This does not occur when running such a test in isolation, which makes me think that something is statically cached. 

I suppose someone familiar with the jqwik code may have a hunch about what recent change may have caused this.

**Comment by jlink** [comment]
I’ll have a look at it. Indeed, there have been changes in code that’s being touched in the stack trace.

**Comment by jlink** [comment]
Hopefully fixed in https://github.com/jlink/jqwik/commit/cc02b52d480495a6ee2f8b44f8884378691de93b

**Comment by jlink** [comment]
Fix released in 1.7.0-SNAPSHOT

@mihxil Try with snapshot release. It worked when I did.
